src/emulator/mcardGCN.c

```diff
@@ -263,11 +233,11 @@ static bool mcardReplaceFileBlock(MemCard* pMCard, s32 index) {
     s32 checksum1;
     s32 checksum2;
     char buffer[0x2000];
-    s32 pad;
+    s32 pad[2];
 
     memcpy(buffer, pMCard->writeBuffer, 0x2000);
 
-    if (!UnkInlinemCardReplaceFileBlock(pMCard)) {
+    if (!mcardReadAnywhereNoTime(pMCard, 0x4000, 0x2000, pMCard->writeBuffer)) {
```

Comment:
```suggestion
    if (!mcardReadAnywhereNoTime(pMCard, 0x4000, CARD_SYSTEM_BLOCK_SIZE, pMCard->writeBuffer)) {
```
maybe that 0x4000 is ``CARD_SYSTEM_BLOCK_SIZE * 2``? not sure if that makes sense though